services/management/ia/templates_service.go

```diff
@@ -49,15 +53,23 @@ const (
 	ruleFileDir = "/tmp/ia1/"
 )
 
+// Rule represents alerting rule/rule template with added source field.
+type Rule struct {
+	alert.Rule
+	Yaml   string
+	Source iav1beta1.TemplateSource
+}
+
 // TemplatesService is responsible for interactions with IA rule templates.
 type TemplatesService struct {
 	db                   *reform.DB
 	l                    *logrus.Entry
 	builtinTemplatesPath string
 	userTemplatesPath    string
+	rulesFileDir         string
```

Comment:
why this field? I don't see it being used in anyway for now. even in the tests you are just creating a temp dir and setting this field but what is the dir used for?

Reply:
I just remembered  its supposed to be used in `dumpRule` here `	path := ruleFileDir + alertRule.Alert + ".yml"`

Reply:
Before that field tests dumped rule files in default dir. These files were caught by running managed instance it devcontainer and that caused the error. (It can't parse this format yet).
So I added that field to force tests to dump files in some temporary directory to eliminate side effects.